src/FormCollection/Form/UXCollectionType.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\FormCollection\Form;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @final
+ * @experimental
+ */
+class UXCollectionType extends AbstractType
+{
+    public function getParent()
+    {
+        return CollectionType::class;
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $defaultButtonAddOptions = [
+            'label' => 'Add',
+            'class' => '',
+        ];
+        $defaultButtonDeleteOptions = [
+            'label' => 'Remove',
+            'class' => '',
+        ];
+        $resolver->setDefaults([
+            'button_add_options' => $defaultButtonAddOptions,
+            'button_delete_options' => $defaultButtonDeleteOptions,
+        ]);
+
+        $resolver->setAllowedTypes('button_add_options', 'array');
+        $resolver->setAllowedTypes('button_delete_options', 'array');
+
+        $resolver->setNormalizer('button_add_options', function (Options $options, $value) use ($defaultButtonAddOptions) {
+            $value['label'] = $value['label'] ?? $defaultButtonAddOptions['label'];
+            $value['class'] = $value['class'] ?? $defaultButtonAddOptions['class'];
+
+            return $value;
+        });
+        $resolver->setNormalizer('button_delete_options', function (Options $options, $value) use ($defaultButtonDeleteOptions) {
+            $value['label'] = $value['label'] ?? $defaultButtonDeleteOptions['label'];
+            $value['class'] = $value['class'] ?? $defaultButtonDeleteOptions['class'];
+
+            return $value;
+        });
```

Comment:
I think we should set `'by_reference' => true` to override the parent `CollectionType`. Forgetting this is a big "WTF" moment. And so, since we're introducing this field new, let's use this "better" default setting.